corehq/apps/locations/templates/locations/manage/location.html

```diff
@@ -74,6 +74,18 @@
       <br />
       {% endif %}
 
+      {% if creates_new_location and not form.is_valid and request.method == 'POST' %}
```

Comment:
Might want to target this a little more precisely - can it check whether any organization levels already exist?  This could be confusing if the error is something else.